python/lsst/resources/_resourcePath.py

```diff
@@ -1428,6 +1428,38 @@ def _openImpl(self, mode: str = "r", *, encoding: str | None = None) -> Iterator
         if "r" not in mode or "+" in mode:
             self.write(out_bytes, overwrite=("x" not in mode))
 
+    def generate_presigned_get_url(self, *, expiration_time_seconds: int) -> str:
+        """Return a pre-signed URL that can be used to retrieve this resource
+        using an HTTP GET without supplying any access credentials.
+
+        Parameters
+        ----------
+            expiration_time_seconds : `int`
+                Number of seconds until the generated URL is no longer valid
+
+        Returns
+        -------
+            url : `str`
+                HTTP URL signed for GET
```

Comment:
```suggestion
                HTTP URL signed for GET.
```
There are other cases of docstrings below.